### ISSUE TITLE
Furnace recipe handler NullPointer

### ISSUE THREAD

**Issue by chazwarp923** [mezz/JustEnoughItems#358]
I was testing my furnace recipes with JEI installed in my dev environment and i noticed that for some odd reason the furnace module of the vanilla integration fails to load due to a NullPointerException in vanilla code. I'm not sure if this is due to how you're calling it or if it truly is a Forge/Vanilla bug but I figured I'd report it. I first discovered it on JEI version 3.7.5.229 but i have tested it to still fail on 230, my dev environment is using Forge 1.10.2-12.18.1.2011 which as of July 13th, 2016 is the latest recommended version. You can find the stacktrace here: http://pastebin.com/RkHzJzgT and i have also attached a compiled version of my mod to see if that helps. Thanks for your time.

[UnifiedItems.zip](https://github.com/mezz/JustEnoughItems/files/362458/UnifiedItems.zip)

**Comment by KnightMiner** [comment]
Looks like someone added a recipe without setting the experience value, if you are adding the recipes make sure you are setting that (such as by using the Forge method)

**Comment by chazwarp923** [comment]
GameRegistry.addSmelting(block.getValue(), new ItemStack(UIItems.ingots.get(block.getKey())), 0.5f);
Thats one of my lines that adds a smelting recipe, i have the 0.5f in the function call

**Comment by mezz** [comment]
Looks like your `stack.getItem()` is returning null, so you are not creating a valid itemstack. 
This may be because your block has no associated Item when you register its smelting recipe.

**Comment by chazwarp923** [comment]
How would that work though? Cause when I test the recipe it works just fine

**Comment by mezz** [comment]
Are you testing the right recipe? How many do you add?

**Comment by chazwarp923** [comment]
I'm loading up mc right now to test them all, but if one works they should all work since i add them all through one for loop

**Comment by mezz** [comment]
Testing myself, I already found a smelting recipe that looks like

`item.dustIron -> null`

**Comment by mezz** [comment]
https://github.com/chazwarp923/Unified-Items/blob/master/src/main/java/tech/chazwarp923/unifieditems/crafting/Smelting.java

In here you should check that your `UIItems.ingots.get` result is not null before making the recipe.

**Comment by chazwarp923** [comment]
Okay, after testing that one i see what i did wrong, but i still think that you should maybe add a case so that it doesn't fail to load the whole module if it gets a null ItemStack

**Comment by mezz** [comment]
Yes, I am now

**Comment by chazwarp923** [comment]
Well if thats all sorted ill just close this

**Comment by mezz** [comment]
Fixed in JEI version 3.7.6. 
Now it will log errors like this instead of crashing the module:

```
[19:22:57] [Client thread/ERROR] [JEI]: Found a broken recipe: mezz.jei.plugins.vanilla.furnace.SmeltingRecipe@1597087b
Output ItemStacks: Found an itemStack with a null item. This is an error from another mod.
Output Fluids: []
Input ItemStacks: [[1xitem.dustIron@0 unifieditems:dustIron]
]
Input Fluids: []

[19:22:57] [Client thread/ERROR] [JEI]: Found a broken recipe: mezz.jei.plugins.vanilla.furnace.SmeltingRecipe@17586565
Output ItemStacks: Found an itemStack with a null item. This is an error from another mod.
Output Fluids: []
Input ItemStacks: [[1xitem.dustGold@0 unifieditems:dustGold]
]
Input Fluids: []

...
```
